sdk/mobile-center-crashes/src/main/java/com/microsoft/azure/mobile/crashes/Crashes.java

```diff
@@ -546,33 +545,54 @@ private void processPendingErrors() {
         }
 
-        if (shouldStopProcessingPendingErrors()) {
-            return;
+        /* If automatic processing is enabled. */
+        if (mAutomaticProcessing) {
+
+            /* Proceed to check if user confirmation is needed. */
+            sendCrashReportsOrAwaitUserConfirmation();
         }
 
-        processUserConfirmation();
     }
 
-    private void processUserConfirmation() {
+    /**
+     * Check user confirmation or send logs now.
+     */
+    private boolean sendCrashReportsOrAwaitUserConfirmation() {
 
         /* Handle user confirmation in UI thread. */
+        final boolean alwaysSend = StorageHelper.PreferencesStorage.getBoolean(PREF_KEY_ALWAYS_SEND, false);
         HandlerUtils.runOnUiThread(new Runnable() {
 
             @Override
             public void run() {
-                boolean shouldAwaitUserConfirmation = true;
-                if (mUnprocessedErrorReports.size() > 0 &&
-                        (StorageHelper.PreferencesStorage.getBoolean(PREF_KEY_ALWAYS_SEND, false)
-                                || !(shouldAwaitUserConfirmation = mCrashesListener.shouldAwaitUserConfirmation()))) {
-                    if (!shouldAwaitUserConfirmation) {
-                        MobileCenterLog.debug(LOG_TAG, "CrashesListener.shouldAwaitUserConfirmation returned false, continue sending logs");
+
+                /* If we still have crashes to send after filtering. */
+                if (mUnprocessedErrorReports.size() > 0) {
+
+                    /* Check for always send: this bypasses user confirmation callback. */
+                    if (alwaysSend) {
+                        MobileCenterLog.debug(LOG_TAG, "The flag for user confirmation is set to ALWAYS_SEND, will send logs.");
+                        handleUserConfirmation(SEND);
+                        return;
+                    }
+
+                    /* For disabled automatic processing, we don't call listener. */
+                    if (!mAutomaticProcessing) {
+                        MobileCenterLog.debug(LOG_TAG, "Automatic processing disabled, will wait for explicit user confirmation.");
+                        return;
+                    }
+
+                    /* Check via listener if should wait for user confirmation. */
+                    if (!mCrashesListener.shouldAwaitUserConfirmation()) {
+                        MobileCenterLog.debug(LOG_TAG, "CrashesListener.shouldAwaitUserConfirmation returned false, will send logs.");
+                        handleUserConfirmation(SEND);
                     } else {
-                        MobileCenterLog.debug(LOG_TAG, "The flag for user confirmation is set to ALWAYS_SEND, continue sending logs");
+                        MobileCenterLog.debug(LOG_TAG, "CrashesListener.shouldAwaitUserConfirmation returned true, wait sending logs.");
                     }
-                    handleUserConfirmation(SEND);
                 }
             }
         });
+        return alwaysSend;
```

Comment:
Why are you returning `alwaysSend`? If it returns `true` when it processes crashes, there is one case that is returning `false` when `mCrashesListener.shouldAwaitUserConfirmation()` returns `false`.

Reply:
We need to return that so that React Native knows that value is persisted into storage, see: https://github.com/Microsoft/mobile-center-sdk-react-native/blob/9de1aa7d0edb00fd2a2ee3bfc320c2b3971a9d39/mobile-center-crashes/Crashes.js#L89

Reply:
This is why we need java doc for this method.
https://github.com/Microsoft/mobile-center-sdk-android/pull/526#discussion_r144171645
Can you just say here that the method returns an alwaysSend flag?

Reply:
I see.